material/src/main/java/org/odk/collect/material/MaterialPill.kt

```diff
@@ -0,0 +1,75 @@
+package org.odk.collect.material
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.ColorInt
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.core.content.res.ResourcesCompat
+import com.google.android.material.shape.MaterialShapeDrawable
+import com.google.android.material.shape.ShapeAppearanceModel
+import org.odk.collect.androidshared.system.ContextUtils.getThemeAttributeValue
+
+/**
+ * Implementation of "pills" present on the Material 3 website and its examples, but not
+ * included in the spec or in Android's MaterialComponents. The pill will use the
+ * `?shapeAppearanceCornerSmall` shape appearance for the current theme.
+ */
+class MaterialPill(context: Context, attrs: AttributeSet?) :
+    FrameLayout(context, attrs) {
+
+    var text: String? = null
+        set(value) {
+            field = value
+            findViewById<TextView>(R.id.text).text = text
+        }
+
+    private val shapeAppearanceModel =
+        ShapeAppearanceModel.builder(context, getShapeAppearance(context), -1).build()
+
+    init {
+        inflate(context, R.layout.pill, this)
+        background = createMaterialShapeDrawable(getDefaultBackgroundColor(context))
+    }
+
+    fun setText(@StringRes id: Int) {
+        text = context.getString(id)
+    }
+
+    fun setIcon(@DrawableRes id: Int) {
+        val drawable = ResourcesCompat.getDrawable(resources, id, context.theme)
+        findViewById<ImageView>(R.id.icon).setImageDrawable(drawable)
```

Comment:
Can we use binding like in other custom vies for example the [CrashView](https://github.com/getodk/collect/blob/master/crash-handler/src/main/java/org/odk/collect/crashhandler/CrashView.kt)?